### ISSUE TITLE
textlint: support --rule ./path/to/rule.js 

### ISSUE THREAD

**Issue by azu** [textlint/textlint#560]
We want to support file path to rule file for command line flag and textlinrc.

## Purpose

It help to debug developing rule.

In my usecase, check the developing rule with actual texts.

```
cat example.md | xargs ./node_modules/textlint/bin/textlint.js --rule lib/my-rule.js -f pretty-error
```

In my real usecase: test my use with [some corpus](https://github.com/textlint-ja/technological-book-corpus-ja).

```
technological-book-corpus-ja | xargs ./node_modules/textlint/bin/textlint.js --rule lib/my-rule.js -f pretty-error
```
## Task

- [ ] support file path in `.textlinrc`
- [ ] support file path in `--rule` flag

```
{
   "rules": {
     "./path/to/rule.js": true
   }
}
```

I don't know that the above case is safe?

Maybe, Other implementation is suitable for this purpose.

**Comment by matthew-piziak** [comment]
Agreed. On systems like NixOS there is more than one path to point at so `--rules-base-directory` doesn't cut it.

**Comment by MartinBonner** [comment]
Sob!  I came here, wanting a solution to the statement in https://github.com/textlint/textlint/blob/master/docs/faq/failed-to-load-textlints-module.md :

> - If you have installed textlint as --global(-g), must install each rule as --global.
> - If you have installed textlint as --save-dev(-D), must install each rule as --save-dev.

It sounds like this would be that solution - but it's not nearly implemented.
